### PR TITLE
config: rm gems jeckyll-redirect-from; update url

### PR DESCRIPTION
**Pull request by brymz**
Attempt at #299.

I mostly guessed at what needed to be changed in this file by what I could remove (gems) and update (url) without breaking `jekyll serve`.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ permalink:        pretty
 title:            Data Carpentry -for- Biologists
 tagline:          'A Jekyll theme'
 description:      'Teaching biologists the tools they need to use computers to do cool science'
-url:              http://www.progbio.org
+url:              http://www.datacarpentry.org
```
Comment (ethanwhite) [inline _config.yml]: I think this should be `http://www.datacarpentry.org/semester-biology`

File: _config.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ permalink:        pretty
 title:            Data Carpentry -for- Biologists
 tagline:          'A Jekyll theme'
```
Comment (ethanwhite) [inline _config.yml]: We should get rid of this.

**Comment by ethanwhite** [comment]
And you're right about being able to drop the redirect-from.

File: _config.yml (hunk 1 above)
Comment (brymz) [inline _config.yml]: Done.

File: _config.yml (hunk 2 above)
Comment (brymz) [inline _config.yml]: Done.

**Comment by brymz** [comment]
Ready to go.

**Pull request merged**